sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/FormRecognizerAsyncClient.java

```diff
@@ -382,18 +384,50 @@ public FormRecognizerServiceVersion getServiceVersion() {
 
     private Function<PollingContext<OperationResult>, Mono<OperationResult>> receiptStreamActivationOperation(
         Flux<ByteBuffer> data, long length, FormContentType formContentType, boolean includeTextDetails) {
-        return (pollingContext) -> {
-            try {
-                Objects.requireNonNull(data, "'data' is required and cannot be null.");
-                Objects.requireNonNull(formContentType, "'formContentType' is required and cannot be null.");
-                return service.analyzeReceiptAsyncWithResponseAsync(ContentType.fromString(formContentType.toString()),
-                    data, length, includeTextDetails)
-                    .map(response -> new OperationResult(
+        Objects.requireNonNull(data, "'data' is required and cannot be null.");
+
+        // Size of buffered data, it used to collect all data from input Flux<ByteBuffer>
+        final long[] bufferedDataSize = {0};
+        // Cached byte buffer
+        final List<ByteBuffer> cachedBuffers = new LinkedList<>();
+        // A boolean value that only used once for guessing content type on input data
+        final AtomicBoolean[] isGuessed = {new AtomicBoolean(true)};
+        // Content type on input data
+        final ContentType[] contentType = new ContentType[1];
+
+        return pollingContext -> data.filter(ByteBuffer::hasRemaining)
+            .windowUntil(buffer -> {
+
+                if (bufferedDataSize[0] >= length) {
+                    return false;
+                }
+                bufferedDataSize[0] += buffer.remaining();
+
+                // if content type is not given, do guessing the content type.
+                if (formContentType == null && isGuessed[0].get() && bufferedDataSize[0] >= 4) {
+                    byte[] bytes = buffer.array();
+                    contentType[0] = getContentType(ByteBuffer.wrap(bytes));
+                    isGuessed[0].compareAndSet(true, false);
+                }
+                // Cache the data to a list of byte buffers
+                ByteBuffer cachedBuffer = ByteBuffer.allocate(buffer.remaining()).put(buffer);
+                cachedBuffer.flip();
+                cachedBuffers.add(cachedBuffer);
+
+                if (bufferedDataSize[0] >= length) {
+                    return true;
+                }
```

Comment:
If the buffer data is greater than length, we should stop and throw an exception as the length provided by user is incorrect.